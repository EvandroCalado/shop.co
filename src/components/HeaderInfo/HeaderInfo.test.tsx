import { render, screen } from '@testing-library/react';
import { HeaderInfo } from '.';

describe('<HeaderInfo />', () => {
  it('should render with default values', () => {
    render(<HeaderInfo />);

    expect(screen.getByText(/get 20% off/i)).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: 'Sign Up Now' }),
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Sign Up Now' })).toHaveAttribute(
      'href',
      '/register',
    );
  });

  it('should render a snapshot', () => {
    const { container } = render(<HeaderInfo />);

    expect(container).toMatchSnapshot();
  });
});
