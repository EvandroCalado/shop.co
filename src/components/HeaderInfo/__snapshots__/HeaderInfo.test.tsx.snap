// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HeaderInfo /> > should render a snapshot 1`] = `
<div>
  <div
    class="flex h-10 w-full items-center justify-center gap-2 bg-black/95 px-4 text-[11px] text-white sm:text-sm"
  >
    <h4>
      Sing up and get 20% off to your first order.
    </h4>
    <a
      class="border-b font-bold duration-150 hover:brightness-90"
      href="/register"
    >
      Sign Up Now
    </a>
  </div>
</div>
`;
